app/main.py

```diff
@@ -2,5 +2,19 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def inner(*args) -> str:
```

Comment:
Use a decorator from `functools` to preserve the wrapped function name and docstring